### DISCUSSION TITLE
part-ii/2015-94/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#196] [Question comments]
Part II 2015 Number Theory: Paper 4, Section II, H 

https://questions.tripos.org/part-ii/2015-94/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Theorem: (Chinese Remainder Theorem) Let $m_1, \dots, m_n$ be pairwise coprime positive integers, and $a_1, \dots, a_n \in \mathbb{Z}$. Then the system of congruences $x \equiv a_i \pmod{m_i}$ is soluble, and the solution is unique $\mod M$, where $M = m_1 \cdot m_2 \cdot \dots \cdot m_n$.

The Jacobi symbol $(\frac{a}{N})$ is defined to be 

$$\prod_{i=1}^{k} (\frac{a}{p_i})$$

where this is a product of Legendre symbols and $N = p_1 \dots p_k$ is a factorisation into primes (not necessarily distinct).

(i) This is false. A counterexample is $(\frac{10}{21}) = (\frac{2}{3})(\frac{5}{7}) = (-1)(-1) = +1$; we cannot have $x^2 \equiv 10 \pmod{21}$, since then we would have $x^2 \equiv 3 \pmod{7}$, but $(\frac{3}{7}) = -1$, so this is impossible.

(ii) This is true. It follows from the Chinese Remainder Theorem; for $N$ an odd prime, the claim is true because we have the same number of quadratic residues as quadratic non-residues. Otherwise, if $N$ is the product of several distinct primes $p_1, p_2, \dots, p_k$, then if we fix $a \pmod{p_i}$ for $2 \leq i \leq k$, then we can take out a constant factor since Jacobi symbols are multiplicative in the denominator, and by the Chinese Remainder Theorem, what we are left with is $\sum_{a=1}^{N} (\frac{a}{p_1})$, which we know equals $0$.

(iii) This is false. Counterexamples are called Carmichael numbers, of which the smallest is $N = 561 = 3 \cdot 11 \cdot 17$. To see that $a^{N-1} \equiv 1 \pmod{N}$ for $(a,N) = 1$, we use Fermat's Little Theorem,  and the fact that $p-1$ divides $N-1$ for each prime $p$ dividing $N$. Then $a^{N-1} \equiv 1 \pmod{3}$, $a^{N-1} \equiv 1 \pmod{11}$, and $a^{N-1} \equiv 1 \pmod{17}$, so $a^{N-1} \equiv 1 \pmod{3 \cdot 11 \cdot 17}$ (for uniqueness we can use the Chinese Remainder Theorem here).

(iv) This is true. The proof is bookwork, but I'll reproduce it anyway.

We split into two cases, based on whether or not $N$ is squarefree.

If N is squarefree, then write $N = pm$, where $p$ is a prime and $m \geq 3$ is an odd integer not divisible by $p$. Then choose $u \in \mathbb{N}$ such that $(\frac{u}{p}) = -1$. By the Chinese Remainder Theorem, there exists $a \in \mathbb{N}$ such that $a \equiv u \pmod{p}$ and $a \equiv 1 \pmod{m}$. Then

$$(\frac{a}{N}) = (\frac{a}{p})(\frac{a}{m}) = (\frac{u}{p})(\frac{1}{m}) = (-1)(+1) = -1$$

but $a^{\frac{N-1}{2}} \equiv 1 \pmod{m}$. Since $m \geq 3$, $-1 \not\equiv +1 \pmod{m}$, so the congruence $a^{\frac{N-1}{2}} \equiv (\frac{a}{N})$ does not hold $\mod m$, so it certainly cannot hold $\mod N$.

On the other hand, if $N$ is not squarefree, then we prove that there exists $a$ coprime to $N$ such that $a^{N-1} \not\equiv 1 \pmod{N}$ (i.e. we prove that $N$ is not a Carmichael number). This implies our desired result by squaring $a^{\frac{N-1}{2}} \equiv (\frac{a}{N}) \pmod{N}$ (i.e. an Euler pseudoprime to the base $a$ is also a Fermat pseudoprime to the base $a$).

Suppose $p$ is a prime such that $p^2$ divides $N$. By the Chinese Remainder Theorem, there exists $a$ such that $a \equiv 1 + p \pmod{p^2}$ and $(a,N) = 1$ (e.g. we can take $L$ to be $N$ divided by the highest power of $p$ that divides it and ask for $a \equiv 1 \pmod{L}$, and then apply the Chinese Remainder Theorem since $L$ is coprime to $p^2$).

Then $a^{N-1} \equiv 1 + (N-1)p \pmod{p^2}$, so $a^{N-1} \not\equiv 1 \pmod{p^2}$, so certainly $a^{N-1} \not\equiv 1 \pmod{N}$, so we are done.

</details>
